src/main/java/com/bookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,79 @@
+package com.bookstore.controller;
+
+import com.bookstore.dto.book.BookResponseDtoWithoutCategoryIds;
+import com.bookstore.dto.category.CategoryRequestDto;
+import com.bookstore.dto.category.CategoryResponseDto;
+import com.bookstore.service.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Category controller", description = "Endpoints for managing books categories")
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(value = "/categories")
+public class CategoryController {
+
+    private final CategoryService categoryService;
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping
+    @Operation(summary = "Create new category", description = "Creates a new book category")
+    public CategoryResponseDto createCategory(
+            @RequestBody @Valid CategoryRequestDto categoryRequestDto) {
+        return categoryService.save(categoryRequestDto);
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN') or hasRole('ROLE_USER')")
```

Comment:
when we use `hasRole`, we don't write prefix **ROLE_**. you can use `hasAnyRole` instead `or`